### PR TITLE
Initial commit for Rotation class

### PR DESCRIPTION
**Pull request by adbugger**
This pull request is for testing the workflow.

### CODE REVIEW

File: scipy/spatial/transform/rotation.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+
+class Rotation(object):
+    """
+    This class implements implements rotations in 3 dimensions using quaternions.
```
Comment (nmayorov) [inline scipy/spatial/transform/rotation.py]: Double "implements" :) 

Also it is more standard to put the summary line on the same line as `"""`. 

As a rule of thumb I usually try to remove unnecessary words. In this case for example you can simply write: "Rotation in 3 dimensions".

Also I believe our concept is that a user doesn't necessary need to know how it is implemented. So in my opinion it is better to mention quaternions somewhere in `Notes` section.

**Comment by nmayorov** [comment]
This approach seems to work well. Only I think you don't need to stick to `rotation_dev` name. You are better to name dev branches based on what you are working.

**Comment by adbugger** [comment]
I changed the docstring. I'll keep creating branches as I work on individual functions and code. This was just a start. Subject to your review, I'll merge this into rotation, and base future branches off of that.

**Comment by nmayorov** [comment]
Also I suggest using "commit tags" as proposed in numpy development guide, please check it https://docs.scipy.org/doc/numpy-1.14.0/dev/gitwash/development_workflow.html ("Writing the commit message").

I will merge this pull request to change how it works.

**Comment by adbugger** [comment]
I'll keep that in mind. Thanks.

**Pull request merged**